lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -42,30 +42,128 @@ using namespace mlir::NPCOMP::Torch;
 
 static LogicalResult verifyLinalgCompatibleTypes(Operation *op,
                                                  PatternRewriter &rewriter) {
-  // For now, use a small allowlist of types we don't reject.
-  // The main culprit in practice is an unknown dtype
-  // when RefineTypes isn't smart enough to propagate it everywhere.
-  // For tensors, we consider the post-conversion tensor type (this pass is
-  // doing a type conversion).
+  // Check the value tensor is ranked as expected by Linalg.
+  // TODO: Remove this check but use a separate verification pass to verify the
+  // invariants expected by later passes.
   auto isValidLinalgType = [](Type type) {
-    if (auto tensor = type.dyn_cast<ValueTensorType>()) {
-      if (auto rankedTensor =
-              tensor.toBuiltinTensor().dyn_cast_or_null<RankedTensorType>()) {
-        if (BaseMemRefType::isValidElementType(rankedTensor.getElementType()))
-          return true;
-      }
-    }
-    if (type.isa<mlir::FloatType, IntegerType, IndexType>())
-      return true;
-    return false;
+    auto tensor = type.dyn_cast<ValueTensorType>();
+    return !tensor || tensor.toBuiltinTensor().isa<RankedTensorType>();
```

Comment:
toBuiltinTensor can return null. So use dyn_cast_or_null (there is no isa_or_null) and rely on implicit boolean check.